Applications/Utils/ModelPreparation/PartitionMesh/MeshPartitioning.cpp

```diff
@@ -0,0 +1,493 @@
+        {
+            for (unsigned k=0; k<ghost_elem->getNNodes(); k++)
+            {
+                const unsigned node_id = ghost_elem->getNodeIndex(k);
+                if (nodes_reseved[node_id])
+                    continue;
+                nodes_reseved[node_id] = true;
+                partition_nodes.push_back(ghost_elem->getNode(k));
+            }
+        }
+
+        partition_end_node_id[ipart] = partition_nodes.size();
+        // Renumber
+        PetscInt local_id = 0;
+
+        std::vector<unsigned> nodes_local_ids_partition;
+        nodes_local_ids_partition.reserve(nnodes);
+        for (std::size_t i=0; i<nnodes; i++)
+        {
+            nodes_local_ids_partition.push_back(0);
+        }
+
+        for (std::size_t i = partition_start_node_id[ipart];
+                i < end_non_ghost_node_id[ipart]; i++)
+        {
+            const unsigned node_id = partition_nodes[i]->getID();
+            node_global_ids[node_id] = global_node_id_offset;
+            nodes_local_ids_partition[node_id] = local_id;
+            local_id++;
+            global_node_id_offset++;
+        }
+        // Assign local IDs to ghost nodes
+        for (std::size_t i = end_non_ghost_node_id[ipart];
+                i < partition_end_node_id[ipart]; i++)
+        {
+            nodes_local_ids_partition[partition_nodes[i]->getID()] = local_id;
+            local_id++;
+        }
+
+        // Count the number of integer variables used to define non-ghost elements
+        const PetscInt num_non_ghost_elems = partition_regular_elements.size();
+        PetscInt nmb_element_idxs = 3 * num_non_ghost_elems;
+        for (PetscInt j=0; j<num_non_ghost_elems; j++)
+        {
+            nmb_element_idxs += partition_regular_elements[j]->getNNodes();
+        }
+        const PetscInt num_ghost_elems = partition_ghost_elements.size();
+        PetscInt nmb_element_idxs_g = 3 * num_ghost_elems;
+        for (PetscInt j=0; j<num_ghost_elems; j++)
+        {
+            nmb_element_idxs_g += partition_ghost_elements[j]->getNNodes();
+        }
+
+        std::string ipart_str = std::to_string(ipart);
+
+        // Number of active elements
+        const PetscInt offset_e = num_non_ghost_elems + nmb_element_idxs;
+        const PetscInt offset_e_g = num_ghost_elems + nmb_element_idxs_g;
+
+        // Reserved for nodes for high order interpolation
+        const PetscInt num_non_ghost_nodes_extra = 0;
+
+        const PetscInt num_nodes_linear_element =   partition_end_node_id[ipart]
+                - partition_start_node_id[ipart];
+        const PetscInt num_nodes_quadratic_element = 0;
+
+        const PetscInt num_nodes_global_mesh = nnodes;
+        // Reserved for nodes for high order interpolation
+        const PetscInt num_nodes_global_mesh_extra = nnodes;
+
+        // Write configuration data and elements of this partition
+        head[0] = num_nodes_linear_element + num_nodes_quadratic_element;
```

Comment:
It is because of the accumulation of head[10] to head[12] from partition to partition.